distributions/R/scripts/new/header_parser.py

```diff
@@ -0,0 +1,173 @@
+import re
+import argparse
+import ctypes  # used in eval
+import warnings
+from sys import stderr
+from pathlib import Path
+
+from CppHeaderParser import CppHeader, CppMethod, CppVariable
+
+from ctype_translator import mapping
+
+
+param_regex = re.compile(
+    r"""
+    [@\\]param                       # PROPER parameter classifier
+    (?P<mode>(\[(in|out|in,out)])?)  # parameter may be in, out, both or unspecified (def. in)
+    \s*
+    (?P<name>[a-z_][a-z0-9_]*)       # won't match C99's unicode names
+    .*$                              # also capture the description
+    """, re.VERBOSE | re.MULTILINE | re.IGNORECASE
+)
+array_regex = re.compile(r'array\s*of\s*(length|size)\s*(?P<lname>[a-z0-9_]+)')
```

Comment:
Provide set of examples which would explain what these regexps would digest.
Without them next programmer reading your code would need to put an extra effort in understanding what this code is doing

Consider exploting doctests for that:
https://docs.python.org/3/library/doctest.html